presto-main/src/main/java/com/facebook/presto/spiller/BinaryFileSpiller.java

```diff
@@ -130,6 +154,10 @@ public void close()
                     String.format("Failed to delete directory [%s]", targetDirectory),
                     e);
         }
+        finally {
+            operatorSpillContext.updateBytes(-spillerDataSize);
```

Comment:
you could implement `operatorSpillContext.close()` or `operatorSpillContext.setSpillReservation(0)`. This would allow you to remove `spillerDataSize` field from this class and I think it would be a little bit easier to understand.